apps/sports-helsinki/src/domain/search/eventSearch/utils.tsx

```diff
@@ -190,13 +198,21 @@ export const getEventSearchVariables = ({
     }
   };
 
+  const sportsKeywords = [
+    ...new Set( // unique keywords
+      sportsCategories.flatMap(
+        (category) => sportsCategoryData?.[category].keywords ?? []
+      )
+    ),
+  ];
+
   return {
     ...getSearchParam(),
     isFree: isFree || undefined,
     end,
     include,
     keywordOrSet1: SPORT_COURSES_KEYWORDS, // Limit search to sport keywords only
-    keywordOrSet2: [...(keyword ?? [])],
+    keywordOrSet2: [...(keyword ?? []), ...sportsKeywords],
```

Comment:
This is just a thought - but I wonder if there will be either collision (might not matter) or situations where `keywordOrSet1` actually limits `keywordOrSet2` results, ie. will the query be now `keywordOrSet1 AND keywordOrSet2` or `keywordOrSet1 OR keywordOrSet2`. .. So if there's missing keyword "liikunta" from `keywordOrSet1` in the event but included "hiihto" from `keywordOrSet2` (what the user wants) - will the results be empty or all works as intended? So basically, is the `keywordOrSet1` filtering out some results that should be there?

Reply:
Mentioned this thread in [LIIKUNTA-407](https://helsinkisolutionoffice.atlassian.net/browse/LIIKUNTA-407) and added note about evaluating whether the query should be changed.